src/macaron/slsa_analyzer/registry.py

```diff
@@ -331,7 +343,166 @@ def _validate_check_relationship(relationship: Any) -> bool:
+            The id of the check to find transitive children.
+
+        Returns
+        -------
+        list[str]
+            The list of transitive children check id.
+
+        Raises
+        ------
+        CheckNotExistError
+            If `check_id` does not exist in the register.
+
+        CheckCircularDependencyError
+            If  cyclic dependency relationship is found.
+        """
+        visited = []
+        stack = [check_id]
+
+        while stack:
+            current_check_id = stack[-1]
+
+            if current_check_id not in visited:
+                visited.append(current_check_id)
+
+                if current_check_id not in self._all_checks_mapping:
+                    raise CheckNotExistError(f"Check {current_check_id} does not exist.")
+
+                # If this check is not defined in the check relationships mapping, it mean that it
+                # doesn't have any children, hence the default empty dictionary.
+                children = self._check_relationships_mapping.get(current_check_id, {})
+
+                for child in children:
+                    if child not in visited:
+                        stack.append(child)
+                    elif child in stack:
+                        raise CheckCircularDependencyError("cycle nodes detected")
+            else:
+                stack.pop()
+
+        return visited
+
+    def get_transitive_parents(self, child_id: str) -> list[str]:
+        """Return a list of all transitive parent for a check.
+
+        Parameters
+        ----------
+        child_id : str
+            The id of the child check.
+
+        Returns
+        -------
+        list[str]
+            The list of transitive parents check id.
+
+        Raises
+        ------
+        CheckNotExistError
+            If `check_id` does not exist in the register.
+
+        CheckCircularDependencyError
+            If  cyclic dependency relationship is found.
+        """
+        visited = []
+        stack = [child_id]
+
+        while stack:
+            current_check_id = stack[-1]
+
+            if current_check_id not in visited:
+                visited.append(current_check_id)
+
+                current_check = self._all_checks_mapping.get(current_check_id, None)
+                if current_check is None:
+                    raise CheckNotExistError(f"Check {current_check_id} does not exist")
+
+                for relation in current_check.depends_on:
+                    parent = relation[0]
+                    if parent not in visited:
+                        stack.append(parent)
+                    elif parent in stack:
+                        raise CheckCircularDependencyError("cycle nodes detected")
+            else:
+                stack.pop()
+
+        return visited
+
+    def get_final_checks(self, ex_pats: list[str], in_pats: list[str]) -> list[str]:
+        """Return a set of the checks' id to run from exclude and include glob patterns.
```

Comment:
```suggestion
        """Return a set of the check ids to run from the exclude and include glob patterns.
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/254/commits/63b2883b703ee2b96584f00cb34f560f49a87db9